### PR TITLE
docs: Updated README for the ts-react-apollo-client

### PR DESCRIPTION
**Pull request by RinkiyaKeDad**
Have updated the README for the React template. Please let me know what all I can improve more in this README.

### CODE REVIEW

**Comment by machi1990** [comment]
Thanks for the PR @RinkiyaKeDad <3. 

@wtrocki @craicoverflow would one of you want to have a quick look for this one so that @RinkiyaKeDad can make further progress for the remaining templates? Thanks.

File: templates/ts-react-apollo-client/README.md [hunk 1]
```diff
@@ -35,15 +19,41 @@ yarn install
 yarn start:client
 ```
 
+## Dependencies And Tools:
```
Comment (craicoverflow) [inline templates/ts-react-apollo-client/README.md:22]:
```suggestion
## Dependencies And Tools
```

File: templates/ts-react-apollo-client/README.md [hunk 2]
```diff
@@ -35,15 +19,41 @@ yarn install
...
 
 - Generate React components
 
 ```
 yarn generate
 ```
 
-> NOTE: You have to run the above commands on each modification of your business model in the server side. This ensure that the client is kept in sync with changes on the server side. 
\ No newline at end of file
+Running `yarn generate` executes a script which relies on two things, [Graphback CLI](https://graphback.dev/docs/cli/graphback-cli) and [Codegen](https://graphql-code-generator.com/). The Graphback CLI is used to run the `graphback generate` command which executes the generation process that generates the GraphQL server based on your data model. And codegen is used in the `graphql codegen` command which generates code from the GraphQL schema.
```
Comment (craicoverflow) [inline templates/ts-react-apollo-client/README.md:57]: I'd like to see this paragraph formatted a little better, so have a separate paragraph mentioning Graphback CLI and GraphQL Code Generator. Even better would be if they were under two separate headings (GraphQL Document Generation and Code Generation)

Also, a sentence should not begin with "And"

File: templates/ts-react-apollo-client/README.md [hunk 3]
```diff
@@ -35,15 +19,41 @@ yarn install
 yarn start:client
 ```
 
+## Dependencies And Tools:
+
+The following tools and technologies have been used to create this template:
```
Comment (craicoverflow) [inline templates/ts-react-apollo-client/README.md:24]: Great list! You could also include GraphQL Code Generator in this list

File: templates/ts-react-apollo-client/README.md [hunk 4]
```diff
@@ -35,15 +19,41 @@ yarn install
...
+  - The rest of the files are to initialize a standard ReactJs Progressive Web App.
+
+> NOTE: All the files can be edited according to your needs except for those that are generated (no need to edit them as they'll be re-generated anyway).
+
 ## Development and adjustment to schema changes
 
-Client is designed to work only with the sample schema.
-When changing graphback schema you will need to recompile all the changes.
+The client is designed to work only with the sample schema.
+When changing the graphback schema you will need to recompile all the changes.
```
Comment (craicoverflow) [inline templates/ts-react-apollo-client/README.md:49]:
```suggestion
When changing the Graphback schema you will need to recompile all the changes.
```

File: templates/ts-react-apollo-client/README.md (hunk 3 above)
Comment (craicoverflow) [reply templates/ts-react-apollo-client/README.md:24]: It is also missing an entry for React

**Review by craicoverflow — CHANGES_REQUESTED** [review]
Great changes! Thanks so much! I've left some suggestions for you.

File: templates/ts-react-apollo-client/README.md (hunk 2 above)
Comment (RinkiyaKeDad) [reply templates/ts-react-apollo-client/README.md:57]: I need to confirm if I've understood this correctly. So ```graphback generate``` creates the schema in the file "graphback.graphql" and then codegen (```graphql codegen```) uses this created schema to further generate the "generated-types.tsx" file. Right?

Comment (wtrocki) [reply templates/ts-react-apollo-client/README.md:57]: Right. And this is actually really hard for people to understand thus we need it documented - idealy some photo or something.

Comment (RinkiyaKeDad) [reply templates/ts-react-apollo-client/README.md:57]: Thanks for clearing it out. Now that I have a good idea I'll make sure others will be able to understand too :)

**Comment by RinkiyaKeDad** [comment]
@craicoverflow I've made the suggested changes. Please check now.

File: templates/ts-react-apollo-client/README.md [hunk 5]
```diff
@@ -1,26 +1,10 @@
 # Graphback React Sample application
 
-Starter React template using one of the Graphback backend templates
+A starter template for React while using one of the Graphback backend templates.
```
Comment (wtrocki) [inline templates/ts-react-apollo-client/README.md:3]: using Graphback based backend

File: templates/ts-react-apollo-client/README.md [hunk 6]
```diff
@@ -1,26 +1,10 @@
...
 
+## Quickstart Guide
```
Comment (wtrocki) [inline templates/ts-react-apollo-client/README.md:5]: Add note:

Client application have been made for sample model. When making any changes on server client side app will change.

**Review by wtrocki — APPROVED** [review]
One of the best changes I have seen from upstream contributors.

**Comment by RinkiyaKeDad** [comment]
Glad you liked the changes :)
@wtrocki I've added the suggestions in the recent commit.

File: templates/ts-react-apollo-client/README.md [hunk 7]
```diff
@@ -19,16 +19,20 @@ yarn install
...
 
 - [GraphQL](https://graphql.org/): GraphQL is an open-source data query and manipulation language for APIs which was publically released by Facebook in 2015.
 
 - [Apollo Client](https://www.apollographql.com/docs/react/): Apollo Client provides a neat abstraction layer and an interface to interact with the GraphQL server. It allows easier building of UI components that fetch data via GraphQL.
 
+- [React](https://reactjs.org/): React is an open-source JavaScript library for building user interfaces or UI components which is maintained by Facebook and a community of individual developers and companies.
+
 - [Material UI](https://material-ui.com/): Material-UI is an open-source project that features React components that implement Google's Material Design. It has been used for styling this sample React template.
 
+- [Codegen](https://graphql-code-generator.com/): GraphQL Code Generator is a tool that generates code out of your GraphQL schema by analyzing it.
```
Comment (craicoverflow) [inline templates/ts-react-apollo-client/README.md:36]:
```suggestion
- [GraphQL Code Generator](https://graphql-code-generator.com/): GraphQL Code Generator is a tool that generates code out of your GraphQL schema by analyzing it.
```

**Comment by craicoverflow** [comment]
Amazing work!

**Comment by RinkiyaKeDad** [comment]
Thanks @craicoverflow! Will start working on the other READMEs now.

**Pull request merged**